vignettes/estimate_incubation.Rmd

```diff
@@ -0,0 +1,84 @@
+---
+title: "Example to estimate incubation period"
+author: "Flavio Finger"
+date: "`r Sys.Date()`"
+output:
+   rmarkdown::html_vignette:
+     toc: true
+     toc_depth: 2
+vignette: >
+  %\VignetteIndexEntry{estimate_incubation}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, echo = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.width=7,
+  fig.height=5,
+  fig.path="figs-overview/"
+)
+```
+
+
+Example using toy data.
+
+Load environment:
+
+```{r, echo = TRUE, message=FALSE}
+library(linelist)
+library(dplyr)
+library(tidyr)
+library(magrittr)
+library(purrr)
+library(ggplot2)
+library(epitrix)
+```
+
+Make a toy linelist object with several possible exposure dates for each case:
+
+```{r, echo = TRUE}
+onsets <- as.Date("2018-01-01") + base::sample(1:10, 20, replace = TRUE)
+mkexposures <- function(x) x - round(rgamma(sample.int(5, size = 1), shape = 12, rate = 3))
+exposures <- sapply(onsets, mkexposures)
+# exposures <- onsets - round(rgamma(20, shape = 12, rate = 3))
+genders <- c("male", "female")
+gender <- sample(genders, 20, replace = TRUE)
+case_types <- c("confirmed", "probable", "suspected", "not a case",
+                "Confirmed", "PROBABLE", "suspected  ", "Not.a.Case")
+case <- factor(sample(case_types, 20, replace = TRUE))
+toy_data <- data.frame("Date of Onset." = onsets,
+                       "SeX_ " = gender,
+                       "Épi.Case_définition" = case)
+toy_data$exposures <- exposures
+
+x <- toy_data %>%
+  clean_data %>%
+  as_linelist(date_onset = "date_of_onset")
```

Comment:
Currently, nothing is really being done with `date_of_onset` here...

Reply:
yes, I thought of maybe making it possible to use the epivars in linelist to define the columns, but haven't (yet) implemented that